fossunited/foss_hackathon/doctype/foss_hackathon_join_team_request/foss_hackathon_join_team_request.py

```diff
@@ -60,12 +64,31 @@ def reject_other_requests(self):
         requests = frappe.get_all(
             JOIN_TEAM_REQUEST,
             filters={
-                "team": self.team,
+                "hackathon": self.hackathon,
                 "status": "Pending",
                 "reciever_email": self.reciever_email,
+                "name": ["!=", self.name],
             },
+            pluck="name",
         )
         for request in requests:
-            request_doc = frappe.get_doc(JOIN_TEAM_REQUEST, request.name)
+            request_doc = frappe.get_doc(JOIN_TEAM_REQUEST, request)
             request_doc.status = "Rejected"
-            request_doc.save()
+            request_doc.save(ignore_permissions=True)
+
+    def validate_sender_as_member(self):
+        if frappe.permissions.is_system_user(frappe.session.user):
+            return
+
```

Comment:
under what conditions will a system user perform this operation? is this code necessary to handle a testcase or is this really needed for production?

Reply:
Backdoor for when some bugs occur and we need to manually do it for someone. Happened a couple of times before, generally its better to give the system user the ability to do it.

Ideally we will never have to use this

Reply:
let's document this in the method docstring?